restli-server/src/main/java/com/linkedin/restli/server/validation/RestLiValidationFilter.java

```diff
@@ -143,6 +144,11 @@ public RestLiValidationFilter(Collection<String> nonSchemaFieldsToAllowInProject
       }
     }
 
+    if (!shouldValidateOnRequest(requestContext))
```

Comment:
prefer to move this to the first line in this call, then you can remove another `if`.

Reply:
It is unsafe to move this earlier, because we first check shouldValidateOnResponse and if true we construct validating schema and put it in the scratchpad.

If we move this earlier, we will miss doing that and cases where request validation is skipped but response validation is enabled will bomb.

Reply:
oh, I misread the first if, it is `shouldValidateOnResponse`.